### PR TITLE
Build pipeline fixes

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
- Ensures CosmosDb secrets are in KeyVault before web app starts
- Checks and removes PR keyvault if it has been soft-deleted
- Adds parameter to disable AKS

### CODE REVIEW

**Comment by brandonpollett** [comment]
I'm not sure what the exact issue is here, but you might want to look at https://docs.microsoft.com/en-us/azure/app-service/deploy-resource-manager-template#define-dependencies and make sure our dependencies are correct. I know that applying configuration changes end up restarting the services.

**Comment by brendankowitz** [comment]
I was looking at that too. This seems to work for Webapps, but same problem in AKS, happy with a stop-gap to unblock first

File: build/pr-variables.yml [hunk 1]
```diff
@@ -6,3 +6,4 @@ variables:
     DeploymentEnvironmentName: '$(resourceGroupRoot)-$(prNumber)'
     TestEnvironmentName: 'OSS PR$(prNumber)'
     ImageTag: '$(build.BuildNumber)'
+    enableAks: false
```
Comment (brendankowitz) [inline build/pr-variables.yml:9]: I'd like to enable this setting from DevOps eventually

File: samples/templates/default-azuredeploy-docker.json [hunk 2]
```diff
@@ -241,9 +238,25 @@
                     "type": "config",
                     "dependsOn": [
                         "[variables('appServiceResourceId')]",
-                        "[if(variables('deployAppInsights'),concat('Microsoft.Insights/components/', variables('appInsightsName')),resourceId('Microsoft.KeyVault/vaults', variables('serviceName')))]"
+                        "[if(variables('deployAppInsights'),concat('Microsoft.Insights/components/', variables('appInsightsName')),resourceId('Microsoft.KeyVault/vaults', variables('serviceName')))]",
+                        "[resourceId('Microsoft.KeyVault/vaults/secrets', variables('serviceName'), 'CosmosDb--Host')]"
```
Comment (brendankowitz) [inline samples/templates/default-azuredeploy-docker.json:242]: Addresses the issue where the container starts up before the keyvault secrets exist

Comment (LTA-Thinking) [reply samples/templates/default-azuredeploy-docker.json:242]: Does this change also need to be in the non-docker deployment script?

File: build/pr-pipeline.yml [hunk 3]
```diff
@@ -311,8 +317,5 @@ stages:
   - testStu3
   - testR4
   - testR5
-  - testAksR4Cosmos
-  - testAksR4Sql
-  - testAksR4SqlContainer
```
Comment (brendankowitz) [inline build/pr-pipeline.yml:316]: These are preventing the normal cleanup step from running when skipped

Comment (LTA-Thinking) [inline build/pr-pipeline.yml:316]: If enableAks is true the cleanup still needs to wait on these jobs.

Comment (brendankowitz) [reply build/pr-pipeline.yml:316]: Yep, don't have a good solution for this yet. They need to be there, but also they cause the entire step to be skipped at the moment...The AKS builds need some more work

File: samples/templates/default-azuredeploy-docker.json (hunk 2 above)
Comment (brendankowitz) [reply samples/templates/default-azuredeploy-docker.json:242]: Possibly, I haven't done enough testing with the other script yet

File: build/pr-pipeline.yml (hunk 3 above)
Comment (brendankowitz) [reply build/pr-pipeline.yml:316]: Will look at something like this: https://elanderson.net/2020/05/azure-devops-pipelines-depends-on-with-conditionals-in-yaml/

**Pull request merged**